JarRunnerIT fast jar test methods consistently failing on CI runs

I haven't been able to reproduce locally but `JarRunnerIT` test methods `testThatFastJarFormatWorks` and `testThatMutableFastJarWorks` have been failing regularly (across various Java versions) with condition timeout exception in CI runs. To help debug what's going, we recently merged this PR https://github.com/quarkusio/quarkus/pull/9960 which has now turned out to be useful, because a subsequent CI run failure[1] now has the following logs, from a failure of 
```
2020-06-15T19:48:44.8880973Z [ERROR] Tests run: 4, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 333.633 s <<< FAILURE! - in io.quarkus.maven.it.JarRunnerIT
2020-06-15T19:48:44.8882135Z [ERROR] testThatMutableFastJarWorks  Time elapsed: 307.78 s  <<< ERROR!
2020-06-15T19:48:44.8884800Z org.awaitility.core.ConditionTimeoutException: Condition with io.quarkus.maven.it.JarRunnerIT was not fulfilled within 1 minutes.
2020-06-15T19:48:45.0599036Z 	at io.quarkus.maven.it.JarRunnerIT.lambda$testThatMutableFastJarWorks$7(JarRunnerIT.java:150)
2020-06-15T19:48:45.0617135Z 	at
```


```
2020-06-15T19:48:27.1122162Z ####### LOG DUMP ON FAILURE (start) ######
2020-06-15T19:48:27.1124138Z Dumping logs that were generated in target/test-classes/projects/project-classic-console-output-mutable-fast-jar/target/output.log for an operation that resulted in org.awaitility.core.ConditionTimeoutException:
2020-06-15T19:48:27.1139467Z 
2020-06-15T19:48:27.1140673Z /opt/hostedtoolcache/AdoptOpenJDK/1.0.0-releases-8-hotspot-normal-latest/x64/jre/bin/java
2020-06-15T19:48:27.1141211Z -jar
2020-06-15T19:48:27.1141816Z /home/runner/work/quarkus/quarkus/integration-tests/maven/target/test-classes/projects/project-classic-console-output-mutable-fast-jar/target/quarkus-app/quarkus-run.jar
2020-06-15T19:48:27.1142172Z __  ____  __  _____   ___  __ ____  ______ 
2020-06-15T19:48:27.1142615Z  --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
2020-06-15T19:48:27.1143056Z  -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
2020-06-15T19:48:27.1143512Z --\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2020-06-15T19:48:27.1144108Z 2020-06-15 19:43:26,858 ERROR [io.qua.application] (main) Failed to start application: java.lang.RuntimeException: Unable to start HTTP server
2020-06-15T19:48:27.1144450Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:432)
2020-06-15T19:48:27.1144835Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServer(VertxHttpRecorder.java:198)
2020-06-15T19:48:27.1145878Z 	at io.quarkus.deployment.steps.VertxHttpProcessor$openSocket-2064782366.deploy_0(VertxHttpProcessor$openSocket-2064782366.zig:119)
2020-06-15T19:48:27.1146937Z 	at io.quarkus.deployment.steps.VertxHttpProcessor$openSocket-2064782366.deploy(VertxHttpProcessor$openSocket-2064782366.zig:36)
2020-06-15T19:48:27.1147338Z 	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:450)
2020-06-15T19:48:27.1147625Z 	at io.quarkus.runtime.Application.start(Application.java:90)
2020-06-15T19:48:27.1148044Z 	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:91)
2020-06-15T19:48:27.1149633Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:61)
2020-06-15T19:48:27.1187782Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
2020-06-15T19:48:27.1188116Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:106)
2020-06-15T19:48:27.1188601Z 	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
2020-06-15T19:48:27.1188756Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-06-15T19:48:27.1192167Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-06-15T19:48:27.1192337Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-06-15T19:48:27.1192492Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2020-06-15T19:48:27.1192641Z 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:31)
2020-06-15T19:48:27.1192795Z 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:18)
2020-06-15T19:48:27.1192992Z Caused by: java.util.concurrent.ExecutionException: java.lang.LinkageError: loader (instance of  io/quarkus/bootstrap/runner/RunnerClassLoader): attempted  duplicate class definition for name: "io/vertx/core/net/impl/VertxEventLoopGroup"
2020-06-15T19:48:27.1193182Z 	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)
2020-06-15T19:48:27.1193337Z 	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1908)
2020-06-15T19:48:27.1193525Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:403)
2020-06-15T19:48:27.1193676Z 	... 16 more
2020-06-15T19:48:27.1193855Z Caused by: java.lang.LinkageError: loader (instance of  io/quarkus/bootstrap/runner/RunnerClassLoader): attempted  duplicate class definition for name: "io/vertx/core/net/impl/VertxEventLoopGroup"
2020-06-15T19:48:27.1194170Z 	at java.lang.ClassLoader.defineClass1(Native Method)
2020-06-15T19:48:27.1194317Z 	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
2020-06-15T19:48:27.1194470Z 	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:79)
2020-06-15T19:48:27.1194625Z 	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:44)
2020-06-15T19:48:27.1194783Z 	at io.vertx.core.http.impl.HttpServerImpl.<init>(HttpServerImpl.java:67)
2020-06-15T19:48:27.1194938Z 	at io.vertx.core.impl.VertxImpl.createHttpServer(VertxImpl.java:310)
2020-06-15T19:48:27.1195088Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.start(VertxHttpRecorder.java:759)
2020-06-15T19:48:27.1195245Z 	at io.vertx.core.Verticle.start(Verticle.java:66)
2020-06-15T19:48:27.1195394Z 	at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$9(DeploymentManager.java:556)
2020-06-15T19:48:27.1195543Z 	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:366)
2020-06-15T19:48:27.1195697Z 	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
2020-06-15T19:48:27.1195857Z 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
2020-06-15T19:48:27.1196022Z 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
2020-06-15T19:48:27.1196178Z 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
2020-06-15T19:48:27.1196334Z 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
2020-06-15T19:48:27.1196490Z 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2020-06-15T19:48:27.1196631Z 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2020-06-15T19:48:27.1196777Z 	at java.lang.Thread.run(Thread.java:748)
2020-06-15T19:48:27.1196863Z 
2020-06-15T19:48:27.1197553Z 2020-06-15 19:43:26,924 ERROR [io.qua.run.Application] (main) Error running Quarkus application: java.lang.RuntimeException: Failed to start quarkus
2020-06-15T19:48:27.1197716Z 	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:519)
2020-06-15T19:48:27.1197861Z 	at io.quarkus.runtime.Application.start(Application.java:90)
2020-06-15T19:48:27.1197996Z 	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:91)
2020-06-15T19:48:27.1198143Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:61)
2020-06-15T19:48:27.1198383Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
2020-06-15T19:48:27.1198527Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:106)
2020-06-15T19:48:27.1198669Z 	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
2020-06-15T19:48:27.1198807Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-06-15T19:48:27.1198960Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-06-15T19:48:27.1199349Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-06-15T19:48:27.1199534Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2020-06-15T19:48:27.1199797Z 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:31)
2020-06-15T19:48:27.1199983Z 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:18)
2020-06-15T19:48:27.1200892Z Caused by: java.lang.RuntimeException: Unable to start HTTP server
2020-06-15T19:48:27.1201264Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:432)
2020-06-15T19:48:27.1201462Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServer(VertxHttpRecorder.java:198)
2020-06-15T19:48:27.1202086Z 	at io.quarkus.deployment.steps.VertxHttpProcessor$openSocket-2064782366.deploy_0(VertxHttpProcessor$openSocket-2064782366.zig:119)
2020-06-15T19:48:27.1207094Z 	at io.quarkus.deployment.steps.VertxHttpProcessor$openSocket-2064782366.deploy(VertxHttpProcessor$openSocket-2064782366.zig:36)
2020-06-15T19:48:27.1211145Z 	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:450)
2020-06-15T19:48:27.1211466Z 	... 12 more
2020-06-15T19:48:27.1225766Z Caused by: java.util.concurrent.ExecutionException: java.lang.LinkageError: loader (instance of  io/quarkus/bootstrap/runner/RunnerClassLoader): attempted  duplicate class definition for name: "io/vertx/core/net/impl/VertxEventLoopGroup"
2020-06-15T19:48:27.1226700Z 	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)
2020-06-15T19:48:27.1229932Z 	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1908)
2020-06-15T19:48:27.1234288Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:403)
2020-06-15T19:48:27.1234791Z 	... 16 more
2020-06-15T19:48:27.1246042Z Caused by: java.lang.LinkageError: loader (instance of  io/quarkus/bootstrap/runner/RunnerClassLoader): attempted  duplicate class definition for name: "io/vertx/core/net/impl/VertxEventLoopGroup"
2020-06-15T19:48:27.1248677Z 	at java.lang.ClassLoader.defineClass1(Native Method)
2020-06-15T19:48:27.1252637Z 	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
2020-06-15T19:48:27.1260807Z 	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:79)
2020-06-15T19:48:27.1266657Z 	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:44)
2020-06-15T19:48:27.1270904Z 	at io.vertx.core.http.impl.HttpServerImpl.<init>(HttpServerImpl.java:67)
2020-06-15T19:48:27.1272893Z 	at io.vertx.core.impl.VertxImpl.createHttpServer(VertxImpl.java:310)
2020-06-15T19:48:27.1279301Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.start(VertxHttpRecorder.java:759)
2020-06-15T19:48:27.1282013Z 	at io.vertx.core.Verticle.start(Verticle.java:66)
2020-06-15T19:48:27.1286941Z 	at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$9(DeploymentManager.java:556)
2020-06-15T19:48:27.1290827Z 	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:366)
2020-06-15T19:48:27.1296581Z 	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
2020-06-15T19:48:27.1300959Z 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
2020-06-15T19:48:27.1307695Z 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
2020-06-15T19:48:27.1310727Z 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
2020-06-15T19:48:27.1316178Z 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
2020-06-15T19:48:27.1326888Z 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2020-06-15T19:48:27.1336413Z 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2020-06-15T19:48:27.1336912Z 	at java.lang.Thread.run(Thread.java:748)
2020-06-15T19:48:27.1337221Z 
2020-06-15T19:48:27.1337468Z 
2020-06-15T19:48:27.1341616Z ####### LOG DUMP ON FAILURE (end) ######
```

I haven't yet had a chance to analyze this log, so creating an issue to keep track of this.

[1] https://github.com/quarkusio/quarkus/pull/10026/checks?check_run_id=773794492

And this one from a Java 11 run https://github.com/quarkusio/quarkus/pull/10026/checks?check_run_id=773794509:

```
2020-06-15T20:10:25.3571578Z ####### LOG DUMP ON FAILURE (start) ######
2020-06-15T20:10:25.3572767Z Dumping logs that were generated in target/test-classes/projects/project-classic-console-output-fast-jar/target/output.log for an operation that resulted in org.awaitility.core.ConditionTimeoutException:
2020-06-15T20:10:25.3582231Z 
2020-06-15T20:10:25.3583155Z /opt/hostedtoolcache/AdoptOpenJDK/1.0.0-releases-11-hotspot-normal-latest/x64/bin/java
2020-06-15T20:10:25.3583768Z -jar
2020-06-15T20:10:25.3632251Z /home/runner/work/quarkus/quarkus/integration-tests/maven/target/test-classes/projects/project-classic-console-output-fast-jar/target/quarkus-app/quarkus-run.jar
2020-06-15T20:10:25.3681738Z __  ____  __  _____   ___  __ ____  ______ 
2020-06-15T20:10:25.3682889Z  --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
2020-06-15T20:10:25.3683664Z  -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
2020-06-15T20:10:25.3684136Z --\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2020-06-15T20:10:25.3684623Z 2020-06-15 20:05:25,481 ERROR [io.qua.application] (main) Failed to start application: java.lang.RuntimeException: Unable to start HTTP server
2020-06-15T20:10:25.3684828Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:432)
2020-06-15T20:10:25.3684998Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServer(VertxHttpRecorder.java:198)
2020-06-15T20:10:25.3685473Z 	at io.quarkus.deployment.steps.VertxHttpProcessor$openSocket-2064782366.deploy_0(VertxHttpProcessor$openSocket-2064782366.zig:119)
2020-06-15T20:10:25.3685921Z 	at io.quarkus.deployment.steps.VertxHttpProcessor$openSocket-2064782366.deploy(VertxHttpProcessor$openSocket-2064782366.zig:36)
2020-06-15T20:10:25.3686087Z 	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:450)
2020-06-15T20:10:25.3686230Z 	at io.quarkus.runtime.Application.start(Application.java:90)
2020-06-15T20:10:25.3686375Z 	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:91)
2020-06-15T20:10:25.3686526Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:61)
2020-06-15T20:10:25.3686660Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
2020-06-15T20:10:25.3686777Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:106)
2020-06-15T20:10:25.3686913Z 	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
2020-06-15T20:10:25.3687061Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-06-15T20:10:25.3687219Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-06-15T20:10:25.3687385Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-06-15T20:10:25.3687540Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2020-06-15T20:10:25.3687683Z 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:31)
2020-06-15T20:10:25.3687829Z 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:18)
2020-06-15T20:10:25.3688900Z Caused by: java.util.concurrent.ExecutionException: java.lang.LinkageError: loader io.quarkus.bootstrap.runner.RunnerClassLoader @66048bfd attempted duplicate class definition for io.vertx.core.net.impl.VertxEventLoopGroup. (io.vertx.core.net.impl.VertxEventLoopGroup is in unnamed module of loader io.quarkus.bootstrap.runner.RunnerClassLoader @66048bfd, parent loader 'app')
2020-06-15T20:10:25.3689220Z 	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
2020-06-15T20:10:25.3689395Z 	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
2020-06-15T20:10:25.3689564Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:403)
2020-06-15T20:10:25.3689710Z 	... 16 more
2020-06-15T20:10:25.3690394Z Caused by: java.lang.LinkageError: loader io.quarkus.bootstrap.runner.RunnerClassLoader @66048bfd attempted duplicate class definition for io.vertx.core.net.impl.VertxEventLoopGroup. (io.vertx.core.net.impl.VertxEventLoopGroup is in unnamed module of loader io.quarkus.bootstrap.runner.RunnerClassLoader @66048bfd, parent loader 'app')
2020-06-15T20:10:25.3690606Z 	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
2020-06-15T20:10:25.3690751Z 	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
2020-06-15T20:10:25.3690903Z 	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:79)
2020-06-15T20:10:25.3691052Z 	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:44)
2020-06-15T20:10:25.3691190Z 	at io.vertx.core.http.impl.HttpServerImpl.<init>(HttpServerImpl.java:67)
2020-06-15T20:10:25.3691344Z 	at io.vertx.core.impl.VertxImpl.createHttpServer(VertxImpl.java:310)
2020-06-15T20:10:25.3691505Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.start(VertxHttpRecorder.java:759)
2020-06-15T20:10:25.3691657Z 	at io.vertx.core.Verticle.start(Verticle.java:66)
2020-06-15T20:10:25.3691801Z 	at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$9(DeploymentManager.java:556)
2020-06-15T20:10:25.3691947Z 	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:366)
2020-06-15T20:10:25.3692094Z 	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
2020-06-15T20:10:25.3692246Z 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
2020-06-15T20:10:25.3692409Z 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
2020-06-15T20:10:25.3692555Z 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
2020-06-15T20:10:25.3692704Z 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
2020-06-15T20:10:25.3692838Z 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2020-06-15T20:10:25.3692989Z 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2020-06-15T20:10:25.3693138Z 	at java.base/java.lang.Thread.run(Thread.java:834)
2020-06-15T20:10:25.3693221Z 
2020-06-15T20:10:25.3693707Z 2020-06-15 20:05:25,603 SEVERE [io.ver.cor.imp.DeploymentManager] (vert.x-eventloop-thread-1) Undeploy failed: java.lang.IllegalStateException: Already undeployed
2020-06-15T20:10:25.3693886Z 	at io.vertx.core.impl.DeploymentManager$DeploymentImpl.doUndeploy(DeploymentManager.java:672)
2020-06-15T20:10:25.3694047Z 	at io.vertx.core.impl.DeploymentManager$DeploymentImpl.undeploy(DeploymentManager.java:641)
2020-06-15T20:10:25.3694185Z 	at io.vertx.core.impl.DeploymentManager.undeployVerticle(DeploymentManager.java:264)
2020-06-15T20:10:25.3694336Z 	at io.vertx.core.impl.DeploymentManager.undeployAll(DeploymentManager.java:289)
2020-06-15T20:10:25.3694479Z 	at io.vertx.core.impl.VertxImpl.lambda$close$21(VertxImpl.java:547)
2020-06-15T20:10:25.3694619Z 	at io.vertx.core.impl.CloseHooks.run(CloseHooks.java:115)
2020-06-15T20:10:25.3694752Z 	at io.vertx.core.impl.VertxImpl.close(VertxImpl.java:546)
2020-06-15T20:10:25.3695007Z 	at io.quarkus.vertx.core.runtime.VertxCoreRecorder.destroy(VertxCoreRecorder.java:200)
2020-06-15T20:10:25.3695159Z 	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$1.run(VertxCoreRecorder.java:63)
2020-06-15T20:10:25.3695309Z 	at io.quarkus.runtime.StartupContext.runAllInReverseOrder(StartupContext.java:82)
2020-06-15T20:10:25.3695515Z 	at io.quarkus.runtime.StartupContext.close(StartupContext.java:73)
2020-06-15T20:10:25.3695668Z 	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:513)
2020-06-15T20:10:25.3695796Z 	at io.quarkus.runtime.Application.start(Application.java:90)
2020-06-15T20:10:25.3695940Z 	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:91)
2020-06-15T20:10:25.3696080Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:61)
2020-06-15T20:10:25.3696212Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
2020-06-15T20:10:25.3696343Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:106)
2020-06-15T20:10:25.3696485Z 	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
2020-06-15T20:10:25.3696627Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-06-15T20:10:25.3696783Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-06-15T20:10:25.3696954Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-06-15T20:10:25.3697093Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2020-06-15T20:10:25.3697236Z 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:31)
2020-06-15T20:10:25.3697380Z 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:18)
2020-06-15T20:10:25.3697472Z 
2020-06-15T20:10:25.3697954Z 2020-06-15 20:05:25,641 ERROR [io.qua.run.Application] (main) Error running Quarkus application: java.lang.RuntimeException: Failed to start quarkus
2020-06-15T20:10:25.3698120Z 	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:519)
2020-06-15T20:10:25.3698246Z 	at io.quarkus.runtime.Application.start(Application.java:90)
2020-06-15T20:10:25.3698394Z 	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:91)
2020-06-15T20:10:25.3698536Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:61)
2020-06-15T20:10:25.3698667Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
2020-06-15T20:10:25.3698803Z 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:106)
2020-06-15T20:10:25.3698938Z 	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
2020-06-15T20:10:25.3699079Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-06-15T20:10:25.3699237Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-06-15T20:10:25.3702045Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-06-15T20:10:25.3702665Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2020-06-15T20:10:25.3703250Z 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:31)
2020-06-15T20:10:25.3704617Z 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:18)
2020-06-15T20:10:25.3705626Z Caused by: java.lang.RuntimeException: Unable to start HTTP server
2020-06-15T20:10:25.3707074Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:432)
2020-06-15T20:10:25.3709709Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServer(VertxHttpRecorder.java:198)
2020-06-15T20:10:25.3711205Z 	at io.quarkus.deployment.steps.VertxHttpProcessor$openSocket-2064782366.deploy_0(VertxHttpProcessor$openSocket-2064782366.zig:119)
2020-06-15T20:10:25.3713637Z 	at io.quarkus.deployment.steps.VertxHttpProcessor$openSocket-2064782366.deploy(VertxHttpProcessor$openSocket-2064782366.zig:36)
2020-06-15T20:10:25.3714157Z 	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:450)
2020-06-15T20:10:25.3714708Z 	... 12 more
2020-06-15T20:10:25.3723054Z Caused by: java.util.concurrent.ExecutionException: java.lang.LinkageError: loader io.quarkus.bootstrap.runner.RunnerClassLoader @66048bfd attempted duplicate class definition for io.vertx.core.net.impl.VertxEventLoopGroup. (io.vertx.core.net.impl.VertxEventLoopGroup is in unnamed module of loader io.quarkus.bootstrap.runner.RunnerClassLoader @66048bfd, parent loader 'app')
2020-06-15T20:10:25.3723530Z 	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
2020-06-15T20:10:25.3724491Z 	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
2020-06-15T20:10:25.3725993Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:403)
2020-06-15T20:10:25.3726305Z 	... 16 more
2020-06-15T20:10:25.3732574Z Caused by: java.lang.LinkageError: loader io.quarkus.bootstrap.runner.RunnerClassLoader @66048bfd attempted duplicate class definition for io.vertx.core.net.impl.VertxEventLoopGroup. (io.vertx.core.net.impl.VertxEventLoopGroup is in unnamed module of loader io.quarkus.bootstrap.runner.RunnerClassLoader @66048bfd, parent loader 'app')
2020-06-15T20:10:25.3733012Z 	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
2020-06-15T20:10:25.3733832Z 	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
2020-06-15T20:10:25.3735174Z 	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:79)
2020-06-15T20:10:25.3736579Z 	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:44)
2020-06-15T20:10:25.3737775Z 	at io.vertx.core.http.impl.HttpServerImpl.<init>(HttpServerImpl.java:67)
2020-06-15T20:10:25.3738843Z 	at io.vertx.core.impl.VertxImpl.createHttpServer(VertxImpl.java:310)
2020-06-15T20:10:25.3740636Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.start(VertxHttpRecorder.java:759)
2020-06-15T20:10:25.3741321Z 	at io.vertx.core.Verticle.start(Verticle.java:66)
2020-06-15T20:10:25.3742762Z 	at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$9(DeploymentManager.java:556)
2020-06-15T20:10:25.3743820Z 	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:366)
2020-06-15T20:10:25.3745260Z 	at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
2020-06-15T20:10:25.3746795Z 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
2020-06-15T20:10:25.3778990Z 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
2020-06-15T20:10:25.3779421Z 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
2020-06-15T20:10:25.3779601Z 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
2020-06-15T20:10:25.3779886Z 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2020-06-15T20:10:25.3780249Z 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2020-06-15T20:10:25.3780421Z 	at java.base/java.lang.Thread.run(Thread.java:834)
2020-06-15T20:10:25.3780508Z 
2020-06-15T20:10:25.3780566Z 
2020-06-15T20:10:25.3780691Z ####### LOG DUMP ON FAILURE (end) ######
```

```
2020-06-15T20:10:25.3881892Z [ERROR] Tests run: 4, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 345.881 s <<< FAILURE! - in io.quarkus.maven.it.JarRunnerIT
2020-06-15T20:10:25.3893086Z [ERROR] testThatFastJarFormatWorks  Time elapsed: 309.349 s  <<< ERROR!
2020-06-15T20:10:25.3903096Z org.awaitility.core.ConditionTimeoutException: Condition with io.quarkus.maven.it.JarRunnerIT was not fulfilled within 1 minutes.
```

--- a/independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/RunnerClassLoader.java
+++ b/independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/RunnerClassLoader.java
@@ -15,7 +15,6 @@ import java.util.function.Function;
 
 /**
  * Classloader used for production application, using the multi jar strategy
- *
  */
 public class RunnerClassLoader extends ClassLoader {
 
@@ -49,10 +48,6 @@ public class RunnerClassLoader extends ClassLoader {
         //note that for performance reasons this CL does not do parent first delegation
         //although the intention is not for it to be a true isolated parent first CL
         //'delegation misses' where the parent throws a ClassNotFoundException are very expensive
-        Class<?> loaded = findLoadedClass(name);
-        if (loaded != null) {
-            return loaded;
-        }
         if (name.startsWith("java.")) {
             return getParent().loadClass(name);
         }
@@ -61,6 +56,10 @@ public class RunnerClassLoader extends ClassLoader {
             return getParent().loadClass(name);
         }
         synchronized (getClassLoadingLock(name)) {
+            Class<?> loaded = findLoadedClass(name);
+            if (loaded != null) {
+                return loaded;
+            }
             ClassLoadingResource[] resources;
             if (packageName == null) {
                 resources = resourceDirectoryMap.get("");

